source/server/listener_impl.cc

```diff
@@ -412,11 +407,19 @@ void ListenerImpl::initialize() {
   // by resetting the watcher.
   if (workers_started_) {
     // If workers_started_ is true, dynamic_init_manager_ should be initialized by listener manager
-    // directly..
+    // directly.
     dynamic_init_manager_.initialize(init_watcher_);
   }
 }
 
+ListenerImpl::~ListenerImpl() {
+  if (!workers_started_) {
+    // We need to remove the listener_init_target_ handle from parent's initManager(), to unblock
+    // parent's initManager to get ready().
+    listener_init_target_.ready();
```

Comment:
Is there a test that covers this?

Reply:
There is now. I thought there should be some listener_impl_test.cc covers everything.
Seems listener_manager_impl_test.cc makes more sense.